### PR TITLE
4.x: HTTP/2.0 Client trailers support #6544

### LINKED ISSUE #6544: Nima - HTTP/2 Client - trailing headers
Nima HTTP/2 client needs to be able to properly receive and send trailing headers.

Enable the test `io.helidon.nima.tests.integration.http2.client.HeadersTest#trailerHeader`

### PR DESCRIPTION
**Pull request by danielkec**
Fixes #6544

### CODE REVIEW

File: webclient/http1/src/main/java/io/helidon/webclient/http1/Http1ClientResponseImpl.java [hunk 1]
```diff
@@ -113,8 +119,38 @@ public ClientResponseHeaders headers() {
...
+                    .orElseGet(() -> clientConfig.socketOptions().readTimeout());
+
+            if (!this.entityRequested) {
+                throw new IllegalStateException("Trailers requested before reading entity.");
+            }
+
+            try {
+                return ClientResponseTrailers.create(this.trailers.get(timeout.toMillis(), TimeUnit.MILLISECONDS));
+            } catch (InterruptedException e) {
+                throw new IllegalStateException("Timeout " + timeout + " reached while waiting for trailers.", e);
```
Comment (Verdent) [inline webclient/http1/src/main/java/io/helidon/webclient/http1/Http1ClientResponseImpl.java:136]: Incorrect wording. Switch with TimeoutException case :-)

File: webclient/http2/src/main/java/io/helidon/webclient/http2/Http2ClientResponseImpl.java [hunk 2]
```diff
@@ -71,8 +83,35 @@ public ClientResponseHeaders headers() {
...
+                .orElseGet(() -> httpClientConfig.socketOptions().readTimeout());
+
+        if (!this.entityRequested) {
+            throw new IllegalStateException("Trailers requested before reading entity.");
+        }
+
+        try {
+            return ClientResponseTrailers.create(this.responseTrailers.get(timeout.toMillis(), TimeUnit.MILLISECONDS));
+        } catch (InterruptedException e) {
+            throw new IllegalStateException("Timeout " + timeout + " reached while waiting for trailers.", e);
```
Comment (Verdent) [inline webclient/http2/src/main/java/io/helidon/webclient/http2/Http2ClientResponseImpl.java:99]: Incorrect wording. Switch with TimeoutException case :-)

**Review by Verdent — CHANGES_REQUESTED** [review]

File: webclient/http2/src/main/java/io/helidon/webclient/http2/Http2ClientResponseImpl.java (hunk 2 above)
Comment (danielkec) [reply webclient/http2/src/main/java/io/helidon/webclient/http2/Http2ClientResponseImpl.java]: fixed

File: webclient/http1/src/main/java/io/helidon/webclient/http1/Http1ClientResponseImpl.java (hunk 1 above)
Comment (danielkec) [reply webclient/http1/src/main/java/io/helidon/webclient/http1/Http1ClientResponseImpl.java]: fixed

**Review by Verdent — APPROVED** [review]

**Pull request merged**